docs/pages/repo/docs/core-concepts/caching.mdx

```diff
@@ -83,6 +83,85 @@ turbo run build --force
 
 Note that `--force` disables cache reads but does not disable cache writes. If you want to disable cache writes, use the `--no-cache` flag.
 
+## Handling Node.js versions
+
+To account for Node.js versions, use [the `engines` key in package.json](https://docs.npmjs.com/cli/v10/configuring-npm/package-json#engines). Turborepo will see the changes to your `package.json` and miss cache when the field is updated.
+
+## Handling platforms and other arbitrary hash contributors
+
+For advanced use cases, you may need to account for operating systems (OS), architecture, or other external factors that you want to contribute to your hash. You can do this in four quick steps.
+
+### 1. Write an arbitrary file to disk
+
+First, create a script that accounts for the hash contributors that you are interested in. For example, here is a Node.js script that looks at platform and architecture and writes those details to a file (`turbo-cache-key.json`):
+
+```js
+#!/usr/bin/env node
+
+const { writeFileSync } = require('fs');
+const { join } = require('path');
+
+const { platform, arch } = process;
+const file = join(__dirname, '..', 'turbo-cache-key.json');
+const str = JSON.stringify({ platform, arch });
+console.log(`Generating cache key: ${str}`);
+writeFileSync(file, str);
+```
+
+### 2. Add the file to your .gitignore
+
+You won't want to commit this file to source control since it's dependent on environment. Add it to your `.gitignore`:
+
+```txt
+// .gitignore
+
++ turbo-cache-key.json
+```
+
+### 3. Add the file to the hash
+
+Now, make sure that `turbo` is aware of the file by adding it to task inputs. You can do this two ways:
+
+- **For specific tasks**: Include the file in [the `inputs` array](/repo/docs/reference/configuration#inputs) of the task(s)
+
+```json
+{
+  "$schema": "https://turbo.build/schema.json",
+  "pipelines": {
+    "build-for-platforms": {
+      "dependsOn": ["^build"],
+      "inputs": ["**/**", "turbo-cache-key.json"]
+    }
+  }
+}
```

Comment:
We might want to hold off on this recommendation until we have support for the feature to add a single file to the rest of the turbo default inputs. This will alter behavior in an unexpected way since this will include everything (even git ignored) files.

Reply:
I can see I contradicted @mehulkar here haha. I think this is fine to leave, but we should be clear about how this isn't 1:1 with existing behavior if you previously didn't have inputs.

Reply:
not contradicting! I think your point is additive and really good. Maybe something like "note that caveats of `inputs` config apply here if you do this, etc etc"

Reply:
As nice as that would be, we need to document the current state of things. This documentation does bring to light some functionality gaps, and we should fix them, but we need to have documentation in the meantime.

The inputs documentation is linked so those caveats are accounted for. If they aren't, then the `inputs` doc needs to improve.

Reply:
I think this is a little different though. This feels like we're advocating for using `"**/**"` in inputs as a replacement for the default hashing, but it's not going to do what you think it will.

Reply:
Given the current, available API, I don't see us having much of an option. I agree it's not ideal.

Reply:
Agreed, I think we just need to add a comment or a callout or something here to say that (we also should have this feature 🔜 )

Reply:
We'll update the `inputs` doc when that lands. 👍